### ISSUE TITLE
New favicon isn't showing up

### ISSUE THREAD

**Issue by kwalrath** [dart-lang/site-www#2472]
The dart.dev favicon should look exactly like the pub.dev favicon, as of #2470. I'm not seeing that. :(

Looking at the page source, it's from https://dart.dev/assets/shared/dart/icon/64.png, which looks like the old icon. I thought it'd be updated when I took on the new site-shared, where I updated [the icon](https://github.com/dart-lang/site-shared/blob/master/src/_assets/image/dart/logo/64.png) in https://github.com/dart-lang/site-shared/pull/119. Maybe I updated the wrong place, or I need to perform another step.

@chalin, does this sound familiar?

**Comment by chalin** [comment]
AFAICT you did the update correctly. If I build and serve locally from master@HEAD I get the right icon. I checked Travis: the master build of #2470 succeeded and it also deployed successfully (and it's using the right version of site-shared).

After all that, the only thing that comes to mind is to delete Travis caches and trigger a rebuild on the latest non-cron master-branch build.

**Comment by kwalrath** [comment]
Thank you, @chalin! I'll try deleting those Travis caches.

**Comment by parlough** [comment]
Fixed by https://github.com/dart-lang/site-www/pull/2946

**Comment by chalin** [comment]
Hi all! Here's a little background. As you know, there are two asset folders, one named `_assets` and the other `assets`.

Some assets, like the `og:image` must always be referred to using the same path. We do this because we want posts (on FB, Twitter, etc) linking to the `og:image` to have a valid link "forever". Resources like these reside under the `assets` folder, and these assets should be referred to using their static paths (i.e., w/o the file's hash).

On the other hand, resources under `_assets` _are_ ment to be referred to via `{% asset ... %}` (that's why the folder name is "private") so that their compile-time generated names include the file hash.

By design, the favicon image was placed in the `assets` folder and so I suppose that it has the same "forever" constraints as an `og:image`. (It's late here, and my quick searches haven't allowed me to confirm that. I'll try to dig up my old notes soon.)

Although #2946 might make the new icon "appear", this change might cause problems later.

cc @johnpryan

**Comment by chalin** [comment]
Oh, I think that I'm remembering why we did it this way -- at least for the `og:image`: Kathy had started seeing 404s. Well, I guess that checking for 404s is one way to confirm my hypothesis :)

**Comment by chalin** [comment]
Taking a closer look ...

@kwalrath: in #2470 you updated the image at the path `site-shared/src/_assets/image/dart/logo/64.png` ... but the favicon is (well, was) read from `/shared/dart/icon/64.png`. Sorry that I didn't notice that before. You need to update the image in both locations (there are two files for the same image because they serve different purposes and need to be accessed via different URLs -- as was explained in my previous comment).

#2946 should be reverted, and the image at `/shared/dart/icon/64.png` updated. Let me know if that resolves the issue.

**Comment by parlough** [comment]
@chalin Thanks for the thorough investigation around that. I can see how the hashed icon location may cause issues(including limiting caching of it). I must have got lost in navigating the submodule and various symlinks so I never noticed that icon wasn't properly updated. I've gone ahead and opened https://github.com/dart-lang/site-shared/pull/127 and https://github.com/dart-lang/site-www/pull/2967

**Comment by chalin** [comment]
So ... is this really finally fixed?!? 😄

**Comment by kwalrath** [comment]
Not sure. :) See my comment in #2967. (And thanks for all the context, @chalin!)
